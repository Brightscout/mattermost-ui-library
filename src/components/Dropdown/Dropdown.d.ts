import {MenuItemProps} from '@Components/MenuItem/MenuItem';

export interface DropdownProps {
    test?: string;
    className?: string;
    children?: React.ReactNode;
    options: MenuItemProps[];
}
