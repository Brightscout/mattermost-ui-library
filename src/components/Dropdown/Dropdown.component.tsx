import React, {useState} from 'react';

import {Dropdown as RbDropdown} from 'react-bootstrap';

import {MenuItem} from '@Components/MenuItem';

import {DropdownProps} from './Dropdown';
import {StyledButton} from './Dropdown.styles';

/**
 *
 *
 *
 *
 * WIP
 *
 * Styling dropdown button
 *
 * creating function for passing the selected value to the user
 *
 * onClicking the value it should the dropdown should close
 *
 * Refactoring code , remove unused props from d.ts
 *
 *
 *
 *
 *
 *
 */

// For Rendering Custom Toggle
interface CustomToggleProps {
    children?: React.ReactNode;
    onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

// Visible dropdown button
const CustomToggle = React.forwardRef(
    (props: CustomToggleProps, ref: React.Ref<HTMLButtonElement>) => (
        <StyledButton
            ref={ref}
            onClick={props.onClick}
        >
            {props.children}
        </StyledButton>
    ),
);

interface CustomMenuProps {
    children?: React.ReactNode;
    style?: React.CSSProperties;
    className?: string;
    labeledBy?: string;
}

// Customizing dropdown Menu
const CustomMenu = React.forwardRef(
    (props: CustomMenuProps, ref: React.Ref<HTMLDivElement>) => {
        return (
            <div
                ref={ref}
                style={props.style}
                className={props.className}
                aria-labelledby={props.labeledBy}
            >
                <ul className='list-unstyled'>{props.children}</ul>
            </div>
        );
    },
);

export const Dropdown = (props: DropdownProps) => {
    const {options} = props;

    const onMenuItemSelectHandler = (
        e: React.MouseEvent<HTMLLIElement, MouseEvent>,
    ) => {
        const selectedSpan = e.target as HTMLSpanElement;
        setActive(selectedSpan.textContent);
    };
    const [active, setActive] = useState<string | null>('');

    return (
        <RbDropdown>
            <RbDropdown.Toggle as={CustomToggle}>
                {'Dropdown Button'}
            </RbDropdown.Toggle>
            <RbDropdown.Menu as={CustomMenu}>
                {options.map((option) => {
                    return (
                        <MenuItem
                            key={option.label}
                            onClick={onMenuItemSelectHandler}
                            label={option.label}
                            className={active === option.label ? 'active' : ''}
                        />
                    );
                })}
            </RbDropdown.Menu>
        </RbDropdown>
    );
};
