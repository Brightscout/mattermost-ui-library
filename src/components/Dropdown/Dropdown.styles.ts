import {Dropdown, DropdownButton as RBDropdownButton} from 'react-bootstrap';
import styled from 'styled-components';

import {Button} from '@Components/Button';

export const DropdownButton = styled(RBDropdownButton)({});

export const DropdownMenu = styled(Dropdown.Menu)({
    width: '100%',
});

export const DropdownItem = styled(Dropdown.Item)({});

export const StyledButton = styled(Button)({});
