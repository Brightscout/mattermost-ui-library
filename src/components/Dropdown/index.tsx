import React, {useEffect, useState, useRef, useCallback} from 'react';

export type DropdownProps = {
    value: string | null;
    placeholder: string;
    onChange: (newValue: string) => void;
    options: DropdownOptionType[];
    customOption?: DropdownOptionType & {
        onClick: (customOptionValue: string) => void;
    };
    loadingOptions?: boolean;
    disabled?: boolean;
    required?: boolean;
    error?: boolean | string;
    className?: string;
}

const Dropdown = ({
    value,
    placeholder,
    options,
    onChange,
    customOption,
    loadingOptions,
    disabled,
    error,
    required,
    className = '',
}: DropdownProps): JSX.Element => {
    const [open, setOpen] = useState(false);
    const dropdownRef = useRef<HTMLDivElement>(null);

    // Handles closing the popover and updating the value when someone selects an option
    const handleInputChange = useCallback((newOption: DropdownOptionType) => {
        setOpen(false);

        // Trigger onChange only if there is a change in the dropdown value
        if (newOption.value !== value) {
            onChange(newOption.value);
        }
    }, [setOpen, onChange, value]);

    // Handles when someone clicks on the custom option
    const handleCustomOptionClick = useCallback(() => {
        // Update the value on the input to indicate custom option has been chosen
        handleInputChange({
            label: customOption?.label,
            value: customOption?.value as string,
        });

        // Take the action that needs to be taken(only if not already taken) to handle when the user chooses the custom option
        if (customOption?.onClick && customOption.value !== value) {
            customOption.onClick(customOption.value);
        }
    }, [customOption, handleInputChange, value]);

    const getOptions = useCallback(() => (customOption ? [...options, {label: customOption.label, value: customOption.value}] : options), [customOption, options]);

    const getLabel = useCallback((optionValue: string | null) => getOptions().find((option) => option.value === optionValue), [getOptions]);

    // Close the dropdown popover when the user clicks outside
    useEffect(() => {
        const handleCloseDropdown = (e: MouseEvent) => !dropdownRef.current?.contains(e.target as Element) && setOpen(false);

        document.addEventListener('click', handleCloseDropdown);

        return () => document.removeEventListener('click', handleCloseDropdown);
    }, []);

    return (
        <div
            className={`dropdown ${error && 'dropdown--error'} position-relative ${className}`}
            ref={dropdownRef}
        >
            <div
                className={
                    `dropdown__field d-flex align-items-center justify-content-between
                    padding-h-15 border-radius-4 cursor-pointer
                    ${open && 'dropdown__field--open'}
                    ${disabled && 'dropdown__field--disabled cursor-not-allowed'}
                    `
                }
            >
                {placeholder && <label className={`dropdown__field-text dropdown__field-placeholder margin-0 wt-400 ${value && 'dropdown__field-placeholder--shifted'}`}>
                    {placeholder}
                    {required && '*'}
                </label>}
                {value && <p className='dropdown__field-text font-size-14 wt-500 margin-0 margin-top-15 text-ellipsis'>
                    {getLabel(value)?.label || getLabel(value)?.value}
                </p>}
                {loadingOptions ? (
                    <div className='dropdown__loader'/>
                ) : (
                    <i className={`fa fa-angle-down dropdown__field-angle font-size-20 ${open && 'dropdown__field-angle--rotated'}`}/>
                )}
                <input
                    type='checkbox'
                    className='dropdown__field-input cursor-pointer'
                    checked={open}
                    onChange={(e) => setOpen(e.target.checked)}
                    disabled={disabled}
                />
            </div>
            <ul className={`dropdown__options-list channel-bg padding-0 border-radius-4 ${open && 'dropdown__options-list--open'}`}>
                {options.map((option) => (
                    <li
                        key={option.value}
                        onClick={() => !disabled && handleInputChange(option)}
                        className='dropdown__option-item text-ellipsis font-size-14 margin-0 padding-v-10 padding-h-25 cursor-pointer'
                    >
                        {option.label || option.value}
                    </li>
                ))}
                {!options.length && <li className='dropdown__option-item text-ellipsis'>{'Nothing to show'}</li>}
                {customOption && (
                    <li
                        onClick={() => !disabled && handleCustomOptionClick()}
                        className='dropdown__option-item dropdown__custom-option text-ellipsis'
                    >
                        {customOption.label || customOption.value}
                    </li>
                )}
            </ul>
            {typeof error === 'string' && <p className='dropdown__err-text font-size-14 margin-top-5 error-text'>{error}</p>}
        </div>
    );
};

export default Dropdown;
