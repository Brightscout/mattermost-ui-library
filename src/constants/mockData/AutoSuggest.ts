// Mock data for "AutoSuggest" component tests
export const onInputValueChange = jest.fn();
export const onOptionClick = jest.fn();
export const suggestionConfig = {
    suggestions: [{
        label: 'Label 1',
        value: 'Value 1',
    }, {
        label: 'Label 2',
        value: 'Value 2',
    }, {
        label: 'Label 3',
        value: 'Value 3',
    }],
    renderValue: (suggestion: Record<string, string>) => suggestion.label,
};
