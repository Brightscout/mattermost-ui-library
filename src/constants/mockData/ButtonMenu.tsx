import React from 'react';

// Mock data for "ButtonMenu" component tests
export const Icon = <i className='icon icon-globe' />;
export const onBtnClick = jest.fn();
export const buttons = [{
    text: 'Button Text',
    icon: Icon,
    onClick: onBtnClick,
}, {
    text: 'Button Text 2',
    icon: Icon,
    onClick: onBtnClick,
}];
