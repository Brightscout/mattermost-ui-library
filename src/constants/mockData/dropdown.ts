// Mock data for "Dropdown" component
export const onChangeHandler = jest.fn();
export const dropdownOptions = [{
    label: 'Dropdown label 1',
    value: 'Dropdown value 1',
}, {
    label: 'Dropdown label 2',
    value: 'Dropdown value 2',
}, {
    label: 'Dropdown label 3',
    value: 'Dropdown value 3',
}];
export const dropdownPlaceholder = 'Dropdown placeholder';
