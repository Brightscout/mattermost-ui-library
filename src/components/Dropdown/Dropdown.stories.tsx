import React from 'react';
import {Story, Meta} from '@storybook/react';

import {Dropdown} from './Dropdown.component';
import {DropdownProps} from './Dropdown';

export default {
    title: 'Component/Dropdown',
    component: Dropdown,
} as Meta<typeof Dropdown>;

const DropdownTemplate: Story<DropdownProps> = (args) => <Dropdown {...args} />;

// Default
export const Default = DropdownTemplate.bind({});
Default.args = {};

export const PassingOptions = DropdownTemplate.bind({});
PassingOptions.args = {
    options: [{label: 'Option 1'}, {label: 'Option 2'}, {label: 'Option 3'}],
};
