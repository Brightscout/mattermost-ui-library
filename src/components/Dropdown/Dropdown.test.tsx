import React from 'react';
import {mount, ReactWrapper} from 'enzyme';

// Mock data
import {onChangeHandler, dropdownOptions, dropdownPlaceholder} from '@Constants/mockData/dropdown';

import Dropdown, {DropdownProps} from '.';

type DropdownComponentProps = Omit<DropdownProps, 'value' | 'placeholder' | 'onChange' | 'options'> & {
    provideOptions?: boolean;
};

const DropdownComponent = ({provideOptions = true, ...props}: DropdownComponentProps) => (
    <Dropdown
        {...props}
        value={null}
        placeholder={dropdownPlaceholder}
        onChange={onChangeHandler}
        options={provideOptions ? dropdownOptions : []}
    />
);

const clickDropdown = (wrapper: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>, checked: boolean) => wrapper.find('input[type="checkbox"]').simulate('change', {target: {checked}});

describe('Dropdown', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(<DropdownComponent/>);
    });

    afterEach(() => {
        component.unmount();
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should show the loader when "loadingOptions" prop is passed', () => {
        const wrapper = mount(<DropdownComponent loadingOptions={true}/>);
        expect(wrapper.find('.plugin-dropdown__loader').exists()).toBeTruthy();
        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('Should show the error message when "error" prop is passed', () => {
        const errorMsg = 'Error Message';
        const wrapper = mount(<DropdownComponent error={errorMsg}/>);
        expect(wrapper.find('.plugin-dropdown__err-text').text().includes(errorMsg)).toBeTruthy();
        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('Should not show the options when the input isn\'t checked', () => {
        expect(component.find('ul.plugin-dropdown__options-list').hasClass('plugin-dropdown__options-list--open')).toBeFalsy();
    });

    it('Should toggle showing the options when clicked', () => {
        clickDropdown(component, true);
        expect(component.find('.plugin-dropdown__options-list').hasClass('plugin-dropdown__options-list--open')).toBeTruthy();
        expect(component.find('.plugin-dropdown__option-item')).toHaveLength(dropdownOptions.length);
        expect(component).toMatchSnapshot();
        clickDropdown(component, false);
        expect(component.find('ul.plugin-dropdown__options-list').hasClass('plugin-dropdown__options-list--open')).toBeFalsy();
    });

    it('Should show the empty state when "options" are empty', () => {
        const wrapper = mount(<DropdownComponent provideOptions={false}/>);
        clickDropdown(wrapper, true);
        expect(wrapper.find('.plugin-dropdown__option-item')).toHaveLength(1);
        expect(wrapper.find('.plugin-dropdown__option-item').text().includes('Nothing to show')).toBeTruthy();
        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('Should render the customOption correctly', () => {
        const customOptionClickHandler = jest.fn();
        const customOptionConfig = {
            label: 'Custom option label',
            value: 'Custom option value',
            onClick: customOptionClickHandler,
        };
        const wrapper = mount(<DropdownComponent customOption={customOptionConfig}/>);
        expect(wrapper.find('.plugin-dropdown__custom-option').exists()).toBeTruthy();
        wrapper.find('.plugin-dropdown__custom-option').simulate('click');
        expect(customOptionClickHandler).toHaveBeenCalledTimes(1);
        expect(wrapper).toMatchSnapshot();
    });
});
