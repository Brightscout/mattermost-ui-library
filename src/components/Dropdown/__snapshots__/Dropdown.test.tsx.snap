// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Should render correctly 1`] = `
<DropdownComponent>
  <Dropdown
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Dropdown label 1",
          "value": "Dropdown value 1",
        },
        Object {
          "label": "Dropdown label 2",
          "value": "Dropdown value 2",
        },
        Object {
          "label": "Dropdown label 3",
          "value": "Dropdown value 3",
        },
      ]
    }
    placeholder="Dropdown placeholder"
    value={null}
  >
    <div
      className="plugin-dropdown undefined position-relative "
    >
      <div
        className="plugin-dropdown__field d-flex align-items-center justify-content-between
                    padding-h-15 border-radius-4 cursor-pointer
                    false
                    undefined
                    "
      >
        <label
          className="plugin-dropdown__field-text plugin-dropdown__field-placeholder margin-0 wt-400 null"
        >
          Dropdown placeholder
        </label>
        <i
          className="fa fa-angle-down plugin-dropdown__field-angle font-20 false"
        />
        <input
          checked={false}
          className="plugin-dropdown__field-input cursor-pointer"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <ul
        className="plugin-dropdown__options-list channel-bg padding-0 border-radius-4 false"
      >
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 1"
          onClick={[Function]}
        >
          Dropdown label 1
        </li>
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 2"
          onClick={[Function]}
        >
          Dropdown label 2
        </li>
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 3"
          onClick={[Function]}
        >
          Dropdown label 3
        </li>
      </ul>
    </div>
  </Dropdown>
</DropdownComponent>
`;

exports[`Dropdown Should render the customOption correctly 1`] = `
<DropdownComponent
  customOption={
    Object {
      "label": "Custom option label",
      "onClick": [MockFunction] {
        "calls": Array [
          Array [
            "Custom option value",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "value": "Custom option value",
    }
  }
>
  <Dropdown
    customOption={
      Object {
        "label": "Custom option label",
        "onClick": [MockFunction] {
          "calls": Array [
            Array [
              "Custom option value",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "value": "Custom option value",
      }
    }
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            "Custom option value",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    options={
      Array [
        Object {
          "label": "Dropdown label 1",
          "value": "Dropdown value 1",
        },
        Object {
          "label": "Dropdown label 2",
          "value": "Dropdown value 2",
        },
        Object {
          "label": "Dropdown label 3",
          "value": "Dropdown value 3",
        },
      ]
    }
    placeholder="Dropdown placeholder"
    value={null}
  >
    <div
      className="plugin-dropdown undefined position-relative "
    >
      <div
        className="plugin-dropdown__field d-flex align-items-center justify-content-between
                    padding-h-15 border-radius-4 cursor-pointer
                    false
                    undefined
                    "
      >
        <label
          className="plugin-dropdown__field-text plugin-dropdown__field-placeholder margin-0 wt-400 null"
        >
          Dropdown placeholder
        </label>
        <i
          className="fa fa-angle-down plugin-dropdown__field-angle font-20 false"
        />
        <input
          checked={false}
          className="plugin-dropdown__field-input cursor-pointer"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <ul
        className="plugin-dropdown__options-list channel-bg padding-0 border-radius-4 false"
      >
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 1"
          onClick={[Function]}
        >
          Dropdown label 1
        </li>
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 2"
          onClick={[Function]}
        >
          Dropdown label 2
        </li>
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 3"
          onClick={[Function]}
        >
          Dropdown label 3
        </li>
        <li
          className="plugin-dropdown__option-item plugin-dropdown__custom-option text-ellipsis"
          onClick={[Function]}
        >
          Custom option label
        </li>
      </ul>
    </div>
  </Dropdown>
</DropdownComponent>
`;

exports[`Dropdown Should show the empty state when "options" are empty 1`] = `
<DropdownComponent
  provideOptions={false}
>
  <Dropdown
    onChange={[MockFunction]}
    options={Array []}
    placeholder="Dropdown placeholder"
    value={null}
  >
    <div
      className="plugin-dropdown undefined position-relative "
    >
      <div
        className="plugin-dropdown__field d-flex align-items-center justify-content-between
                    padding-h-15 border-radius-4 cursor-pointer
                    plugin-dropdown__field--open
                    undefined
                    "
      >
        <label
          className="plugin-dropdown__field-text plugin-dropdown__field-placeholder margin-0 wt-400 null"
        >
          Dropdown placeholder
        </label>
        <i
          className="fa fa-angle-down plugin-dropdown__field-angle font-20 plugin-dropdown__field-angle--rotated"
        />
        <input
          checked={true}
          className="plugin-dropdown__field-input cursor-pointer"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <ul
        className="plugin-dropdown__options-list channel-bg padding-0 border-radius-4 plugin-dropdown__options-list--open"
      >
        <li
          className="plugin-dropdown__option-item text-ellipsis"
        >
          Nothing to show
        </li>
      </ul>
    </div>
  </Dropdown>
</DropdownComponent>
`;

exports[`Dropdown Should show the error message when "error" prop is passed 1`] = `
<DropdownComponent
  error="Error Message"
>
  <Dropdown
    error="Error Message"
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Dropdown label 1",
          "value": "Dropdown value 1",
        },
        Object {
          "label": "Dropdown label 2",
          "value": "Dropdown value 2",
        },
        Object {
          "label": "Dropdown label 3",
          "value": "Dropdown value 3",
        },
      ]
    }
    placeholder="Dropdown placeholder"
    value={null}
  >
    <div
      className="plugin-dropdown plugin-dropdown--error position-relative "
    >
      <div
        className="plugin-dropdown__field d-flex align-items-center justify-content-between
                    padding-h-15 border-radius-4 cursor-pointer
                    false
                    undefined
                    "
      >
        <label
          className="plugin-dropdown__field-text plugin-dropdown__field-placeholder margin-0 wt-400 null"
        >
          Dropdown placeholder
        </label>
        <i
          className="fa fa-angle-down plugin-dropdown__field-angle font-20 false"
        />
        <input
          checked={false}
          className="plugin-dropdown__field-input cursor-pointer"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <ul
        className="plugin-dropdown__options-list channel-bg padding-0 border-radius-4 false"
      >
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 1"
          onClick={[Function]}
        >
          Dropdown label 1
        </li>
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 2"
          onClick={[Function]}
        >
          Dropdown label 2
        </li>
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 3"
          onClick={[Function]}
        >
          Dropdown label 3
        </li>
      </ul>
      <p
        className="plugin-dropdown__err-text font-14 margin-top-5 error-text"
      >
        Error Message
      </p>
    </div>
  </Dropdown>
</DropdownComponent>
`;

exports[`Dropdown Should show the loader when "loadingOptions" prop is passed 1`] = `
<DropdownComponent
  loadingOptions={true}
>
  <Dropdown
    loadingOptions={true}
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Dropdown label 1",
          "value": "Dropdown value 1",
        },
        Object {
          "label": "Dropdown label 2",
          "value": "Dropdown value 2",
        },
        Object {
          "label": "Dropdown label 3",
          "value": "Dropdown value 3",
        },
      ]
    }
    placeholder="Dropdown placeholder"
    value={null}
  >
    <div
      className="plugin-dropdown undefined position-relative "
    >
      <div
        className="plugin-dropdown__field d-flex align-items-center justify-content-between
                    padding-h-15 border-radius-4 cursor-pointer
                    false
                    undefined
                    "
      >
        <label
          className="plugin-dropdown__field-text plugin-dropdown__field-placeholder margin-0 wt-400 null"
        >
          Dropdown placeholder
        </label>
        <div
          className="plugin-dropdown__loader"
        />
        <input
          checked={false}
          className="plugin-dropdown__field-input cursor-pointer"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <ul
        className="plugin-dropdown__options-list channel-bg padding-0 border-radius-4 false"
      >
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 1"
          onClick={[Function]}
        >
          Dropdown label 1
        </li>
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 2"
          onClick={[Function]}
        >
          Dropdown label 2
        </li>
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 3"
          onClick={[Function]}
        >
          Dropdown label 3
        </li>
      </ul>
    </div>
  </Dropdown>
</DropdownComponent>
`;

exports[`Dropdown Should toggle showing the options when clicked 1`] = `
<DropdownComponent>
  <Dropdown
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Dropdown label 1",
          "value": "Dropdown value 1",
        },
        Object {
          "label": "Dropdown label 2",
          "value": "Dropdown value 2",
        },
        Object {
          "label": "Dropdown label 3",
          "value": "Dropdown value 3",
        },
      ]
    }
    placeholder="Dropdown placeholder"
    value={null}
  >
    <div
      className="plugin-dropdown undefined position-relative "
    >
      <div
        className="plugin-dropdown__field d-flex align-items-center justify-content-between
                    padding-h-15 border-radius-4 cursor-pointer
                    plugin-dropdown__field--open
                    undefined
                    "
      >
        <label
          className="plugin-dropdown__field-text plugin-dropdown__field-placeholder margin-0 wt-400 null"
        >
          Dropdown placeholder
        </label>
        <i
          className="fa fa-angle-down plugin-dropdown__field-angle font-20 plugin-dropdown__field-angle--rotated"
        />
        <input
          checked={true}
          className="plugin-dropdown__field-input cursor-pointer"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <ul
        className="plugin-dropdown__options-list channel-bg padding-0 border-radius-4 plugin-dropdown__options-list--open"
      >
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 1"
          onClick={[Function]}
        >
          Dropdown label 1
        </li>
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 2"
          onClick={[Function]}
        >
          Dropdown label 2
        </li>
        <li
          className="plugin-dropdown__option-item text-ellipsis font-14 margin-0 padding-v-10 padding-h-25 cursor-pointer"
          key="Dropdown value 3"
          onClick={[Function]}
        >
          Dropdown label 3
        </li>
      </ul>
    </div>
  </Dropdown>
</DropdownComponent>
`;
